### ISSUE TITLE
OC 7.x - IIS: Initial creation of calendar fails when user has german langage settings

### ISSUE THREAD

**Issue by art-ist** [owncloud/calendar#603]
In german language the default calendarname is "Persönlich" which contains non ascii caracter.
This causes DB error dur to incorrect UTF translation. The forum link describes the problem and its workaround:
http://forum.owncloud.org/viewtopic.php?p=66704

**Comment by ev0lister** [comment]
Same issue here for me on IIS. What information do you need to fix?

**Comment by georgehrke** [comment]
@ev0lister please use the issue template

**Comment by georgehrke** [comment]
and the calendar doesn't support ISS

**Comment by georgehrke** [comment]
at least we don't aim to support ISS

**Comment by art-ist** [comment]
The source of the problem seems to be the poor Unicode-support of PHP for windows, not an actual bug in the calendar code itself.
**But the solution would be simple:**
As the calendar actually works on a Windows/IIS based installation. The main problem relies on the non ascii characters in the default name for the personal calendar. 
A simple **change of the default calendar name** for new users from 'Pers**ö**nich' to **'Privat'** (=private ... wich would be a sufficient translation) **or 'Mein Kalender'** (=my calendar) would be a reasonable and very simple workaround that would leave most users to never run into any troubles when using the calendar module on a windows based ownCloud installation.
@georgehrke: this solution wouldn't require any windows testing machine on your side as well.

**Comment by georgehrke** [comment]
Translation is not a code issue and we don't translate on github. If you want to change the translation of a string, you can contribute at https://www.transifex.com/projects/p/owncloud/

**Comment by cmeh** [comment]
First of all thanks for reporting this issue with Microsoft IIS, @art-ist.

From a translator's perspective, "Persönlich" certainly remains the better, more literal translation of "Personal". The way I understand the discussion above is that moving away from it for the purpose of working around bugs on non-targeted platforms (in this case Microsoft's IIS) is rather against the consensus here.

So the most elegant solution could IMHO be a short reference note here on what to patch if you really want to host ownCloud on the not targeted Windows IIS.

For reference: The corresponding resources on Transifex are
* https://www.transifex.com/projects/p/owncloud/translate/#de/calendar/1812074 (German - "Du"),
* https://www.transifex.com/projects/p/owncloud/translate/#de_DE/calendar/1812074 (German - "Sie"),
* https://www.transifex.com/projects/p/owncloud/translate/#de_AT/calendar/1812074 (Austria) and
* https://www.transifex.com/projects/p/owncloud/translate/#de_CH/calendar/1812074 (Switzerland).

In the .tar.bz2 respectively .zip ownCloud archive downloaded for server installation, just by briefly looking over the files on a Debian-based server *(please feel free to correct me if I'm wrong!)* , these would correspond to …

`root@RaspberryPi /var/www/owncloud # grep Persönlich core/l10n/*.js*`
`de_AT.js:    "Personal" : "Persönlich",`
`de_AT.json:    "Personal" : "Persönlich",`
`de_CH.js:    "Personal" : "Persönlich",`
`de_CH.json:    "Personal" : "Persönlich",`
`de_DE.js:    "Personal" : "Persönlich",`
`de_DE.json:    "Personal" : "Persönlich",`
`de.js:    "Personal" : "Persönlich",`
`de.json:    "Personal" : "Persönlich",`

So in other words, it shouldn't be too difficult to exchange "Persönlich" with "Privat" by the means of scripting, either by using Microsoft's PowerShell, a batch file or whatever comes to your own mind. This could be a rather limited, additional task you'd perform once on ownCloud's installation and then once on each subsequent ownCloud update.

**Comment by ev0lister** [comment]
Thanks for this explanation. Will report back to PHP Windows Team. Anyhow two comments

@georgehrke It's really IIS not ISS (international space station..)

And secondly, it's understandable that "you dont aim for Windows", but think a bit for the future of Owncloud- there might be a point where a customer will have the need to run it on Windows for whatever reasons. As you are providing a business version, this should be kept in mind.

Thanks for all your replys.

**Comment by georgehrke** [comment]
>  As you are providing a business version, this should be kept in mind.

The calendar is a completely community-driven app and has absolutely nothing todo with the enterprise edition offered by owncloud Inc.

**Comment by art-ist** [comment]
@ev0lister and @georgehrke thanks for your effort.
Isn't the whole idea of the ownCloud project to empower people to keep control over their (personal) data? Wouldn't that include a huge amount of people who do not happen to be highly skilled Linux admins or PHP devs?
ownCloud has that power and if a solution could be relativley easy to implement, it shouldn't be a problem if some of the devs don't have a windows box to test. I there should definitley be some guys in the community to be able to do some testing on a windows box. Could that be a solution?

**Comment by georgehrke** [comment]
I said that __I__ don't care about IIS, that doesn't mean __other people can__ not send pull request fixing this issue.
And at the risk of repeating myself: this bug tracker is not the right place for translations. ownCloud is localised via transfex. If I change the translation directly in this repo, it will be overwritten by the translation bot within the next 24hours.
If you want to change the translation, please get an account at transfex and change it there.

**Comment by cmeh** [comment]
@georggehrke Message from the translators: The above-mentioned resources are partly in a "reviewed" state, so @art-ist won't be able to change them (even if equipped with a newly created Transifex account).

But even if we un-reviewed these resources to make them changeable again, the remaining question behind all that would be whether bugs on certain platforms should be superficially “resolved” by modifying translations away from the original choice to a translation farer away from the English term just to work around these bugs.

So, as Georg has already pointed out, pull requests from people with an in-depth knowledge of IIS are highly appreciated here, and even if they won't be available immediately, scripting can provide a workaround for the meantime.
